src/pages/_app.js

```diff
@@ -0,0 +1,55 @@
+import Head from 'next/head';
+import { CacheProvider } from '@emotion/react';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+import { CssBaseline } from '@mui/material';
+import { ThemeProvider } from '@mui/material/styles';
+import { AuthConsumer, AuthProvider } from 'src/contexts/auth-context';
+import { useNProgress } from 'src/hooks/use-nprogress';
+import { createTheme } from 'src/theme';
+import { createEmotionCache } from 'src/utils/create-emotion-cache';
+import 'simplebar-react/dist/simplebar.min.css';
+
+const clientSideEmotionCache = createEmotionCache();
+
+const SplashScreen = () => null;
+
+const App = (props) => {
+  const { Component, emotionCache = clientSideEmotionCache, pageProps } = props;
+
+  useNProgress();
+
+  const getLayout = Component.getLayout ?? ((page) => page);
+
+  const theme = createTheme();
+
+  return (
+    <CacheProvider value={emotionCache}>
+      <Head>
+        <title>
+          Devias Kit
+        </title>
+        <meta
+          name="viewport"
+          content="initial-scale=1, width=device-width"
+        />
+      </Head>
+      <LocalizationProvider dateAdapter={AdapterDateFns}>
+        <AuthProvider>
+          <ThemeProvider theme={theme}>
```

Comment:
This could cause issues if we want to switch back to the app router I mentioned in the other comment. I think this has to be client component. But app router defaults to server components. I think using 'use client' will force them to be client rendered but could get messy too.